vunit/sim_if/xcelium.py

```diff
@@ -0,0 +1,694 @@
+        )
+        group.add_argument(
+            "--cdslib",
+            default=None,
+            help="The cds.lib file to use. If not given, VUnit maintains its own cds.lib file.",
+        )
+        group.add_argument(
+            "--hdlvar",
+            default=None,
+            help="The hdl.var file to use. If not given, VUnit does not use a hdl.var file.",
+        )
+        group.add_argument(
+            "--use_indago",
+            default=False,
+            help="Indicate whether to use Indago instead of Simvision when running in GUI mode.",
+            action="store_true",
+        )
+
+    @classmethod
+    def from_args(cls, args, output_path, **kwargs):
+        """
+        Create new instance from command line arguments object
+        """
+        return cls(
+            prefix=cls.find_prefix(),
+            output_path=output_path,
+            log_level=args.log_level,
+            gui=args.gui,
+            cdslib=args.cdslib,
+            hdlvar=args.hdlvar,
+            use_indago=args.use_indago
+        )
+
+    @classmethod
+    def find_prefix_from_path(cls):
+        """
+        Find xcelium simulator from PATH environment variable
+        """
+        return cls.find_toolchain(["xrun"])
+
+
+    @staticmethod
+    def supports_vhdl_contexts():
+        """
+        Returns True when this simulator supports VHDL 2008 contexts
+        """
+        return True
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self, prefix, output_path, gui=False, log_level=None, cdslib=None, hdlvar=None, use_indago=False
+    ):
+        SimulatorInterface.__init__(self, output_path, gui)
+        self._global_compile_options = {}
+        self._prefix = prefix
+        self._libraries = []
+        self._log_level = log_level
+        if cdslib is None:
+            self._cdslib = str((Path(output_path) / "cds.lib").resolve())
+        else:
+            self._cdslib = str(Path(cdslib).resolve())
+        self._hdlvar = hdlvar
+        self._use_indago = use_indago
+        self._cds_root_xrun = self.find_cds_root_xrun()
+
+
+    def find_cds_root_xrun(self):
+        """
+        Finds xrun cds root
+        """
+        return subprocess.check_output(
+            [str(Path(self._prefix) / "cds_root"), "xrun"]
+        ).splitlines()[0].decode()
+
+    def find_cds_root_virtuoso(self):
+        """
+        Finds virtuoso cds root
+        """
+        try:
+            return subprocess.check_output(
+                [str(Path(self._prefix) / "cds_root"), "virtuoso"],
+                stderr=subprocess.STDOUT
+            ).splitlines()[0].decode()
+        except subprocess.CalledProcessError:
+            return None
+
+    def _create_cdslib(self):
+        """
+        Create the cds.lib file in the output directory if it does not exist
+        """
+        cds_root_virtuoso = self.find_cds_root_virtuoso()
+
+        if cds_root_virtuoso is None:
```

Comment:
I've never understood driving TCL interfaces with Python strings. Stuff like this is way clearer when using Jinja templates as the flow can be followed in a single file with all the for, if/else, etc. constructs instead of following python strings between functions/methods

Reply:
For multi-line strings, I like using `inspect.cleandoc` instead to avoid `\`'s and also to keep the string inline with the code

Reply:
Is Jinja already a VUnit dependency? If so, I agree, otherwise it's not quite so clear if that code warrants an extra dependency.
I didn't know about `inspect.cleandoc`, interesting. I often see `textwrap.dedent` used (assuming the tabs --> spaces transformation is not needed).

Reply:
If nothing else, this should use f-strings.
I think I was using Python 3.2 when I originally wrote this which apparently didn't even support `{}` with auto-numbering.

Reply:
I'm not sure, it's been a while since I've forked and modified VUnit. `textwrap` probably makes more sense, I know `cleandoc` because I've needed some other stuff it can do that `dedent` can't.

Reply:
Ideally, I'd just convert this to format strings. But I don't see a need to change it unless people are very unhappy with it.